app/assets/javascripts/admin/addon/components/site-settings/file-size-restriction.hbs

```diff
@@ -0,0 +1,11 @@
+<FileSizeInput
+  @sizeValueKB={{readonly this.value}}
+  @onChangeSize={{this.onChangeSize}}
+  @updateValidationMessage={{this.updateValidationMessage}}
+  @min={{if this.setting.min this.setting.min null}}
+  @max={{if this.setting.max this.setting.max null}}
+  @message={{this.validationMessage}}
+/>
+
+<SettingValidationMessage @message={{this.validationMessage}} />
+<div class="desc">{{html-safe this.setting.description}}</div>
```

Comment:
This component is small enough that it may make sense to build it in `.gjs`, template and JS all in one file.